browser/ui/webui/new_tab_page/instant_service_message_handler.cc

```diff
@@ -237,3 +241,61 @@ void InstantServiceMessageHandler::HandleSetMostVisitedSettings(
     instant_service_->ToggleMostVisitedOrCustomLinks();
   }
 }
+
+void InstantServiceMessageHandler::HandleEditTopSite(
+    const base::ListValue* args) {
+  AllowJavascript();
+
+  std::string url;
+  if (!args->GetString(0, &url))
+    return;
+
+  std::string new_url;
+  if (!args->GetString(1, &new_url))
+    return;
+
+  std::string title;
+  if (!args->GetString(2, &title)) {
+    title = url;
+  }
+
+  // Normalize url if invalid.
+  // TODO(simonhong): It's very simple. Use more better way.
+  if (!new_url.empty() && !GURL(new_url).is_valid())
```

Comment:
I think the validation logic is fine (it matches behavior in Chrome) - but maybe pull this validation out to a method so that you can test it easily 😄

Reply:
Done!